addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1181,6 +1181,141 @@ QUnit.test('Post a message containing an email address followed by a mention on
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+        document.execCommand('insertText', false, "#");
+        document.querySelector(`.o_ComposerTextInput_textarea`)
+            .dispatchEvent(new window.KeyboardEvent('keydown'));
+        document.querySelector(`.o_ComposerTextInput_textarea`)
+            .dispatchEvent(new window.KeyboardEvent('keyup'));
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_ComposerSuggestion').click()
+    );
+    await afterNextRender(() => {
+        document.querySelector('.o_Composer_buttonSend').click();
+    });
+    assert.containsOnce(
+        document.querySelector('.o_Message_content'),
+        '.o_channel_redirect',
+        "message should contain a link the the mentionned channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_channel_redirect').textContent,
+        '#General & good',
+        "link to the channel must containt # + the channel name"
+    );
+});
+
+QUnit.test('mention a channel and add a new line after the mention', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records.push({
+        id: 7,
+        name: "General good",
+    });
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 7,
+        model: 'mail.channel',
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+        document.execCommand('insertText', false, "#");
+        document.querySelector(`.o_ComposerTextInput_textarea`)
+            .dispatchEvent(new window.KeyboardEvent('keydown'));
+        document.querySelector(`.o_ComposerTextInput_textarea`)
+            .dispatchEvent(new window.KeyboardEvent('keyup'));
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_ComposerSuggestion').click()
+    );
+    await afterNextRender(() => {
+        document.execCommand('insertText', false, "\n test");
+    });
+    await afterNextRender(() => {
+        document.querySelector('.o_Composer_buttonSend').click();
+    });
+    assert.containsOnce(
+        document.querySelector('.o_Message_content'),
+        '.o_channel_redirect',
+        "message should contain a link the the mentionned channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_channel_redirect').textContent,
+        '#General good',
+        "link to the channel must containt # + the channel name"
+    );
+});
```

Comment:
I just realized I gave you the wrong setup for this test, and the one I intended to cover is actually still not working.

First here is the correct setup for this test to cover the fix you did. I'm pretty sure this test is currently succeeding even if you revert the `\n` fix.

It should be changed so that there is a first line that contains a `#` but it is not a mention.
And then the mention should be on the 2nd line (this would be similar to the existing partner test).



And now the tests I had in mind, that are still not working (neither for partner or channel):

Have a mention directly followed by a `\n`, by **replacing** the space that is automatically added after selecting a mention by a `\n`. This bug is due to using space as separator being really weak.

(And a similar version but instead of a \n it's the end of the input).

Basically `_computeMentionedPartners` and `_computeMentionedChannels`, instead of looking for things that look like mentions with regex and then matching them to channel/partner name, should look for known partner/channel name (+ the mention starting char) inside the text input.

Reply:
> Have a mention directly followed by a \n, by replacing the space that is automatically added after selecting a mention by a \n. This bug is due to using space as separator being really weak.

This case is not fixed by the current code. I'll build the test and try to find a solution.